spec/lib/gitlab_api/base_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'rails_helper'
+
+describe GitlabApi::Base do
+  describe '#connection' do
+    let(:gitlab_api_url) { 'https://gitlab.com/api/v4' }
+    let(:gitlab_private_token) { 'asdfasdf' }
+
+    before do
+      allow(api).to receive(:gitlab_api_url).and_return(gitlab_api_url)
+      allow(api).to receive(:gitlab_private_token).and_return(gitlab_private_token)
+    end
+
+    let(:api) { described_class.new }
+    subject { api.connection }
+
+    it { is_expected.to be_a(Faraday::Connection) }
+
+    it 'creates a connection with proper params' do
+      is_expected.to have_attributes(
+        headers: hash_including(
+          'Content-Type' => 'application/json',
+          'Accept' => 'application/json',
+          'Private-Token' => gitlab_private_token
+        ),
+        url_prefix: be_a(URI::HTTPS) & have_attributes(to_s: gitlab_api_url)
```

Comment:
would be better verify the content instead the type in `url_prefix: be_a(URI::HTTPS)`